Content.Server/Doors/Systems/AirlockSystem.cs

```diff
@@ -109,5 +113,60 @@ private void OnDoorPry(EntityUid uid, AirlockComponent component, BeforeDoorPryE
                 args.Cancel();
             }
         }
+
+        private void OnRangedInteract(EntityUid uid, AirlockComponent component, RangedInteractEvent args)
+        {
+            args.Handled = true;
+            var interactionSystem = EntityManager.EntitySysManager.GetEntitySystem<SharedInteractionSystem>();
+            if (!interactionSystem.InRangeUnobstructed(args.UserUid, args.TargetUid, -1f, CollisionGroup.Opaque))
+            {
+                args.Handled = false;
+                return;
```

Comment:
just in case there was something that already handled this event first, this should start with a 
```cs
if (args.Handled)
	return
```

Also, maybe just set `args.Handled = true` after this and the next two checks succeed. that way you don't have to set it back to false before `return`ing